Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -2097,7 +2097,9 @@ public IEnumerable<TItem> DisplayData
         [Parameter( CaptureUnmatchedValues = true )]
         public Dictionary<string, object> Attributes { get; set; }
 
-        public int SelectedRowIndex => Data.Index( x => x.IsEqual( SelectedRow ) );
+        public int SelectedRowIndex => Virtualize 
+            ? Data.Index( x => x.IsEqual( SelectedRow ) ) 
+            : Data.Index( x => x.IsEqual( SelectedRow ) ) * CurrentPage;

```

Comment:
Isn't it `(CurrentPage - 1) * PageSize + Data.Index(...)` for manual read mode? For in-memory I think page calculation is not needed.

Reply:
/facepalm haha